### PR TITLE
healthcheck_startup_sleep_config

### PR DESCRIPTION
**Pull request by raymondng**
1) Fix "bug" in log ... which says 1 minute when in fact it's 30 seconds :)

2) Make this configurable -- for abook/scraper and perhaps other non-shoebox services, we may not need such a long sleep duration during startup ... in fact we may not need it at all. This is a quick way to test -- I'll first tweak/test this in abook/scraper.

Implicit goal: faster deployment (esp when the # of machines increases as we scale out)

cc: @stkem 
cc: @andrewconner 
cc: @eishay 
cc: @ymatsuda

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala [hunk 1]
```diff
@@ -210,10 +210,12 @@ class HealthcheckPluginImpl @Inject() (
     email
   }
 
+  val sleep = sys.props.getOrElse("healthcheck.startup.sleep", "30").toInt // seconds
```
Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: Could we try to get away from using these sys properties? I've found it confusing to have two locations for configurations, and it's hard to set them in dev mode (so in dev, we're always running defaults basically).

If you find sys properties are helpful for tweaking things, how about a simple `ConfigRepo` where keys can be read first from `sys.props` and then from our TypeSafe configs (which are tracked in source control and support local overrides)?

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: @stkem and @eishay may want to be involved.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: Thanks for the input. I've been thinking the same -- to provide a general & consistent way to support more dynamic means (including MBean-based) of configuration. So, let me look into that and track that in technical debt.

One step at a time though -- as a priority, we should try to remove hard-coded config values (with no config overrides) first  ... we cannot afford to redeploy every time we need to make tweaks. It's fragile and does not scale. I _think_ we are all in agreement about this but do let me know if you feel differently.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: Added tracking task: https://app.asana.com/0/5442988295898/9194036489135
Please feel free to comment.

Comment (andrewconner) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: Our current configuration system (using Typesafe config) doesn't require a redeployment, and supports overrides natively. Using Java system property overrides means we're using two split configuration systems (one of which is a superset of the other), and makes it much harder to track what configuration is being used in production. Tweaking can still be done using Typesafe config the same way, and keeps our configuration unified, tracked, and predictable when you are not tweaking.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: Cool -- sorry I think I mis-read your original comment. 
So, we are on the same page about [1] removing hard-coded config values and [2] utilizing more dynamic config overrides including system properties.
Let me try this out.

Comment (raymondng) [inline kifi-backend/modules/common/app/com/keepit/common/healthcheck/Healthcheck.scala:213]: I don't know if depending on Play.current for simple configuration is a good idea ... as discussed, will revisit later.

**Pull request merged**